src/core/algorithms/order/order.cpp

```diff
@@ -0,0 +1,366 @@
+#include "order.h"
+
+#include <algorithm>
+#include <iostream>
+#include <memory>
+#include <utility>
+
+#include <easylogging++.h>
+
+#include "config/names_and_descriptions.h"
+#include "config/tabular_data/input_table/option.h"
+#include "dependency_checker.h"
+#include "list_lattice.h"
+#include "model/types/types.h"
+#include "order_utility.h"
+
+namespace algos::order {
+
+Order::Order() : Algorithm({}) {
+    RegisterOptions();
+    MakeOptionsAvailable({config::TableOpt.GetName()});
+}
+
+void Order::RegisterOptions() {
+    using namespace config::names;
+    using namespace config::descriptions;
+    using config::Option;
+
+    RegisterOption(config::TableOpt(&input_table_));
+}
+
+void Order::LoadDataInternal() {
+    typed_relation_ = model::ColumnLayoutTypedRelationData::CreateFrom(*input_table_, false);
+}
+
+void Order::ResetState() {}
+
+void Order::CreateSingletonSortedPartitions() {
+    std::vector<model::TypedColumnData> const& data = typed_relation_->GetColumnData();
+    std::unordered_set<unsigned long> null_rows;
+    for (unsigned int i = 0; i < data.size(); ++i) {
+        if (!model::Type::IsOrdered(data[i].GetTypeId()) && !data[i].IsMixed()) {
+            continue;
+        }
+        std::vector<std::byte const*> const& byte_data = data[i].GetData();
+        for (size_t k = 0; k < byte_data.size(); ++k) {
+            if (byte_data[k] == nullptr) {
+                null_rows.insert(k);
+            }
+        }
```

Comment:
As far as I remember, `GetData()` can't contain `nullptr`s and correct way to check for `NULL`s or empty values is to use `IsNull` and `IsEmpty` methods

Reply:
However, the fact that this code works tells us either that I'm wrong and `data` can contain `nullptr`s or that this code wasn't tested on datasets with `NULL`s or empty values :)

Reply:
I'm pretty sure it was tested on datasets with `NULL`s

Reply:
Definetly with empty values